recipes/librtlsdr/all/conanfile.py

```diff
@@ -0,0 +1,130 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+import glob
+
+required_conan_version = ">=1.28.0"
+
+
+class LibRtlSdrConan(ConanFile):
+    name = "librtlsdr"
+    description = "Software to turn the RTL2832U into an SDR"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://osmocom.org/projects/rtl-sdr/wiki/Rtl-sdr"
+    license = "GPL-2.0"
+    topics = ("conan", "sdr", "rtl-sdr")
+    generators = ("cmake", "cmake_find_package")
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("libusb/1.0.24")
+        if self.settings.compiler == "Visual Studio":
+            self.requires("pthreads4w/3.0.0")
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = glob.glob(self.name + "-*/")[0]
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.configure(
+            args=["-DCMAKE_BUILD_TYPE={}".format(self.settings.build_type), "-DCMAKE_VERBOSE_MAKEFILE=ON"],
+            source_folder=self._source_subfolder,
+            build_folder=self._build_subfolder,
+        )
```

Comment:
- build_type is automatically handled
- we never enable verbose makefiles
- don't add `source_folder`. This bypasses your own cmake build script.
```suggestion
        self._cmake.configure(build_folder=self._build_subfolder)
```

Reply:
This cmake recipe is poorly made and I noticed a log message on it saying "build type not set, defaulting to release", that's why I added that. I'm going to test as you suggest.

Reply:
The verbose file was to see why this isn't building on MacOS. I'll remove it when I debbug it, okay?

Reply:
Okay, just tested on my machine and without `source_folder=self._source_subfolder` it doesn't work... I noticed that yeah, most recipes don't have that argument... `cern_root` does, tho.

Why?

Reply:
Almost all conan recipes building with cmake ship a cmake wrapper script.
This wrapper script includes `conanbuildinfo.cmake` (generated by the `cmake` generator) and adds the main project as a subdirectory.

`cern-root` is a "bad" example as it patches the project instead.

See e.g.:
https://github.com/conan-io/conan-center-index/blob/master/recipes/wslay/all/CMakeLists.txt
+
https://github.com/conan-io/conan-center-index/blob/b7b5b0f0cc586510d946aa68f6a496a905a3a1d5/recipes/wslay/all/conanfile.py#L26

Reply:
> The verbose file was to see why this isn't building on MacOS. I'll remove it when I debbug it, okay?

Okay. Note that instead of passing `-DCMAKE_VERBOSE_MAKEFILE=ON`. you can do `self._cmake.verbose = True`.
That will pass the `--verbose` flag to `cmake --build`.

Reply:
I'll try that approach. Maybe it'll even help solve the horrible libusb detection haha. Thanks!

Reply:
```
-- Found: C:/Users/ruilv/.conan/data/libusb/1.0.24/_/_/package/3fb49604f9c2f729b85ba3115852006824e72cab/lib/libusb-1.0.lib
-- Looking for pthread.h
-- Looking for pthread.h - not found
-- Found Threads: TRUE
CMake Error: File C:/Users/ruilv/.conan/data/librtlsdr/0.8/_/_/build/0fecb8a3e41a65c7ca2229ccf9432fbd31f334d1/cmake/cmake_uninstall.cmake.in does not exist.
CMake Error at source_subfolder/CMakeLists.txt:147 (configure_file):
  configure_file Problem configuring file


-- Udev rules not being installed, install them with -DINSTALL_UDEV_RULES=ON
-- Building with kernel driver detaching disabled, use -DDETACH_KERNEL_DRIVER=ON to enable
CMake Error at source_subfolder/src/CMakeLists.txt:43 (include):
  include could not find requested file:

    C:/Users/ruilv/.conan/data/librtlsdr/0.8/_/_/build/0fecb8a3e41a65c7ca2229ccf9432fbd31f334d1/cmake/Modules/Version.cmake


CMake Error at source_subfolder/src/CMakeLists.txt:68 (set_target_properties):
  set_target_properties called with incorrect number of arguments.


CMake Error at source_subfolder/src/CMakeLists.txt:69 (set_target_properties):
  set_target_properties called with incorrect number of arguments.


-- Building for version:  /
```

It breaks apart when trying to add a wrapper CMakeLists. This build code is *horrible*.